File: Parte-2/JavaScript/fibonacci.js
function fibonacci (index=10) {
  if (!index) index = 10
  let sequence = [0, 1]
  for (let i = 2; i <= index; i++) {
    let n = sequence.length - 1 //last index of sequence
    let n_1 = n - 1 //index before the last of sequence
    sequence.push(sequence[n] + sequence[n_1])
  }
  //console.log(sequence)
  console.log(`O ${index}º termo da sequencia é: ${sequence[index]}`)
}

function getExecutionTime(method) { //calc a function execution time
  let args = Array.prototype.slice.call(arguments, 1)
  let startTime = performance.now()
  method.apply(null, args)
  return performance.now() - startTime
}

function init () {
  let arg = parseInt(process.argv[2])
  if (process.argv[2] && !arg) {
    throw new TypeError("O numero passado como argumento deve ser um inteiro.")
  }
  console.log(`Tempo de execucao do fibonacci: ${getExecutionTime(fibonacci, arg)}ms`)
}

init()

File: Parte-2/Python/fibonacci.py
import time
import sys


def fibonacci(ne):
    n = 0
    n_1 = 1
    sequence = [n, n_1]
    for _ in range(0, ne - 1):
        aux = n + n_1
        n = n_1
        n_1 = aux
        sequence.append(aux)
    print(f"O {ne}º termo da sequencia é: {sequence[ne]}")


def preprocess():
    if len(sys.argv) == 2:
        return int(sys.argv[1])
    elif len(sys.argv) != 1:
        print("Qauntidade de argumentos invalida")
        exit()
    else:
        return 10


inicio = time.time()

n = preprocess()
fibonacci(n)

fim = time.time()
print(f"Tempo de execucao do fibonacci: {fim - inicio}")
